arch/arm/boot/dts/overlays/gpio-fan-overlay.dts

```diff
@@ -1,29 +1,37 @@
 /*
  * Overlay for the Raspberry Pi GPIO Fan @ BCM GPIO12.
+ * References: 
+ *	- https://www.raspberrypi.org/forums/viewtopic.php?f=107&p=1367135#p1365084
+ *
  * Optional parameters:
- *	- "gpiopin"	- default GPIO12
- * 	- "temp"	- default 55000
+ *	- "gpiopin"	- BCM number of the pin driving the fan, default 12 (GPIO12);
+ * 	- "temp"	- CPU temperature at which fan is started in millicelsius, default 55000;
+ *
  * Requires:
- *	- kernel configurations: CONFIG_SENSORS_GPIO_FAN=m and CONFIG_SENSORS_PWM_FAN=m;
- *	- kernel rebuid;
- *	- DC Fan connected to GPIO via a N-MOSFET (2N7002)
+ *	- kernel configurations: CONFIG_SENSORS_GPIO_FAN=m;
+ *	- kernel rebuild;
+ *	- N-MOSFET connected to gpiopin, 2N7002-[https://en.wikipedia.org/wiki/2N7000];
+ *	- DC Fan connected to N-MOSFET Drain terminal, a 12V fan is working fine and quite silently;
+ *	  [https://www.tme.eu/en/details/ee40101s1-999-a/dc12v-fans/sunon/ee40101s1-1000u-999/]
  *
  *                   ┌─────────────────────┐
  *                   │Fan negative terminal│
  *                   └┬────────────────────┘
- *                    │
- *                 │──┘
+ *                    │D
+ *             G   │──┘
  * [GPIO12]──────┤ │<─┐  2N7002
  *                 │──┤
- *                    │
+ *                    │S
  *                   ─┴─
  *                   GND
  *
- * sudo dtc -W no-unit_address_vs_reg -@ -I dts -O dtb -o /boot/overlays/gpio-fan.dtbo gpio-fan.dts
- * sudo nano /boot/config.txt add "dtoverlay=gpio-fan" or "dtoverlay=gpio-fan,gpiopin=12,temp=45000"
- * or
- * sudo sh -c "echo '\n# Enable PI GPIO-Fan\ndtoverlay=gpio-fan\n' >> /boot/config.txt"
- * sudo sh -c "echo '\n# Enable PI GPIO-Fan\ndtoverlay=gpio-fan,gpiopin=12\n' >> /boot/config.txt"
+ * Build:
+ * 	- `sudo dtc -W no-unit_address_vs_reg -@ -I dts -O dtb -o /boot/overlays/gpio-fan.dtbo gpio-fan.dts`
```

Comment:
The build instructions are incorrect because the filename is now `gpio-fan-overlay.dts`, but the kernel's `make dtbs` will do the compilation in most cases so this is less important.